plugins/processors/k8sdecorator/stores/utils.go

```diff
@@ -51,42 +53,49 @@ func parseDeploymentFromReplicaSet(name string) string {
 		return ""
 	}
 
-	if !stringInRuneset(suffix, kubeAllowedStringAlphaNums) {
+	if deploymentUnallowedRegExp.MatchString(suffix) {
 		// Invalid suffix
 		return ""
 	}
 
 	return name[:lastDash]
 }
 
-// get the cronJob name by stripping the last dash following some rules
-// return empty if it is not following the rule
+// Get the cronJob name by stripping the last dash following by the naming convention: JobName-UnixTime
+// based on https://github.com/kubernetes/kubernetes/blob/c4d752765b3bbac2237bf87cf0b1c2e307844666/pkg/controller/cronjob/cronjob_controllerv2.go#L594-L596.
+// Before v1.21 CronJob in Kubernetes has used Unix Time in second; after v1.21 is a Unix Time in Minutes.
+
 func parseCronJobFromJob(name string) string {
 	lastDash := strings.LastIndexAny(name, "-")
+
+	//Return empty since the naming convention is: JobName-UnixTime, if it does not have the "-", meanings the job name is empty
 	if lastDash == -1 {
-		// No dash
 		return ""
 	}
+
 	suffix := name[lastDash+1:]
-	if len(suffix) != 10 {
-		// Invalid suffix if it is not 10 rune
+	suffixInt, err := strconv.ParseInt(suffix, 10, 64)
+
+	if err != nil {
 		return ""
 	}
 
-	if !stringInRuneset(suffix, cronJobAllowedString) {
-		// Invalid suffix
+	//Convert Unix Time In Minutes to Unix Time
+	suffixStringMultiply := strconv.FormatInt(suffixInt*60, 10)
+
+	//Checking if the suffix is a unix time by checking: the length and contains character
+
+	//Checking for the length: CronJobControllerV2 is Unix Time in Minutes (7-9 characters) while CronJob is Unix Time (10 characters).
+	//However, multiply by 60 to convert the Unix Time In Minutes back to Unix Time in order to have the same condition as Unix Time
+	if len(suffix) != 10 && len(suffixStringMultiply) != 10 {
+		return ""
```

Comment:
Add tests for this return case

Reply:
These threes are for this return case https://github.com/aws/amazon-cloudwatch-agent/pull/345/files#diff-56aec85db8ce21fd23ac105fdbd3e19b8e7c93ec59777091a68deb2b726e6c59R25-R26, https://github.com/aws/amazon-cloudwatch-agent/pull/345/files#diff-56aec85db8ce21fd23ac105fdbd3e19b8e7c93ec59777091a68deb2b726e6c59R21

Reply:
Can you check again, if you look at the coverage report this return point is never executed.

Reply:
All the unit tests for this now use Unix time. Can you pass a string literal with a suffix that doesn't have a length of 10 instead of depending on the output of the Unix time? That would be a simple way to confirm hitting this return in test coverage.

Reply:
Yes. Added only number test case with suffix smaller than 10 numbers though.